### ISSUE TITLE
ParameterListWrapping works incorrect for params with generics

### ISSUE THREAD

**Issue by kuFEAR** [detekt/detekt#3706]
## Expected Behavior
correct detecting indents for generics
```kotlin
    internal fun provideSomeTooLongNameRepository(
        someTooLongNameRemoteDataSource: SomeTooLongNameRemoteDataSource,
        someTooLongNameLocalDataSource: SomeTooLongNameLocalDataSource,
        someTooLongNameCombiner: SomeTooLongNameCombiner<
            @JvmSuppressWildcards Info,
            @JvmSuppressWildcards SomeTooLongNameInfo>
    ): SomeTooLongNameRepository {
        return SomeTooLongNameRepository(
            someTooLongNameRemoteDataSource,
            someTooLongNameLocalDataSource,
            someTooLongNameCombiner
        )
    }
```

## Observed Behavior
after run task with autoCorrect the result:
```kotlin
    internal fun provideSomeTooLongNameRepository(
    someTooLongNameRemoteDataSource: SomeTooLongNameRemoteDataSource,
    someTooLongNameLocalDataSource: SomeTooLongNameLocalDataSource,
    someTooLongNameCombiner: SomeTooLongNameCombiner<
        @JvmSuppressWildcards Info,
        @JvmSuppressWildcards SomeTooLongNameInfo>
): SomeTooLongNameRepository {
        return SomeTooLongNameRepository(
            someTooLongNameRemoteDataSource,
            someTooLongNameLocalDataSource,
            someTooLongNameCombiner
        )
    }
```

## Steps to Reproduce
enable ParameterListWrapping rule:
```
ParameterListWrapping:
    active: true
    autoCorrect: true
    indentSize: 4
    maxLineLength: 120
```

## Your Environment
* Version of detekt used: 1.16.0
* Version of Gradle used: 7.0
* Operating System and version: osx 11.2.3

**Comment by cortinico** [comment]
`ParameterListWrapping` is a rule that wraps a KtLint rule.
I believe this is potentially related to this https://github.com/pinterest/ktlint/issues/1092 and is fixed on master, as we bumped KtLint to 0.41.0. 

I'm going to close this as it is for now as is not really actionable on our end. If you wish you can try the `master-SNAPSHOT` and verify that the issue is not happening anymore. Feel free to re-open if is not the case 👍
